apps/dolly-backend/src/main/java/no/nav/dolly/bestilling/histark/HistarkClient.java

```diff
@@ -0,0 +1,145 @@
+import java.util.List;
+import java.util.stream.Collectors;
+
+import static java.util.Objects.isNull;
+import static java.util.Objects.nonNull;
+import static no.nav.dolly.domain.resultset.SystemTyper.HISTARK;
+import static org.apache.commons.lang3.StringUtils.isBlank;
+import static org.apache.commons.lang3.StringUtils.isNotBlank;
+
+@Slf4j
+@Service
+@RequiredArgsConstructor
+public class HistarkClient implements ClientRegister {
+
+    private final HistarkConsumer histarkConsumer;
+    private final MapperFacade mapperFacade;
+    private final TransaksjonMappingService transaksjonMappingService;
+    private final ObjectMapper objectMapper;
+    private final TransactionHelperService transactionHelperService;
+    private final PdlPersonConsumer pdlPersonConsumer;
+    private final ErrorStatusDecoder errorStatusDecoder;
+
+    @Override
+    public Flux<ClientFuture> gjenopprett(RsDollyUtvidetBestilling bestilling, DollyPerson dollyPerson, BestillingProgress progress, boolean isOpprettEndre) {
+
+        if (nonNull(bestilling.getHistark())) {
+
+            var bestillingId = progress.getBestilling().getId();
+            return Flux.from(getPersonData(List.of(dollyPerson.getIdent()))
+                            .map(person -> buildRequest(bestilling.getHistark(), person))
+                            .filter(env -> !transaksjonMappingService.existAlready(HISTARK,
+                                    dollyPerson.getIdent(), "NA") || isOpprettEndre)
+                            .flatMap(request -> histarkConsumer.postHistark(request)
+                                    .mapNotNull(status -> getStatus(dollyPerson.getIdent(), bestillingId, status)))
+                            .collect(Collectors.joining(",")))
+                    .map(status -> futurePersist(progress, status));
+        }
+
+        return Flux.empty();
+    }
+
+    @Override
+    public void release(List<String> identer) {
+
+        // Sletting er ikke støttet
+    }
+
+    private ClientFuture futurePersist(BestillingProgress progress, String status) {
+
+        return () -> {
+            transactionHelperService.persister(progress, BestillingProgress::setHistarkStatus, status);
+            return progress;
+        };
+    }
+
+    private String getStatus(String ident, Long bestillingId, HistarkResponse response) {
+
+        if (isNull(response)) {
+            return null;
+        }
+
+        if (isBlank(response.getFeilmelding())) {
+
+            saveTransaksjonId(response.getHistarkId(), ident, bestillingId);
+            return "OK";
+
+        } else {
+
+            return String.format("NA:FEIL=Teknisk feil se logg! %s",
+                    isNotBlank(response.getFeilmelding()) ?
+                            ErrorStatusDecoder.encodeStatus(errorStatusDecoder.getStatusMessage(response.getFeilmelding())) :
+                            "UKJENT");
+        }
+    }
+
+    private Flux<PdlPersonBolk.PersonBolk> getPersonData(List<String> identer) {
```

Comment:
Hvis du ikke trenger annen informasjon enn ident er det vel ingen grunn til å hente opp personen fra PDL?!

Reply:
Nei du har helt rett, dette ble hentet litt for ukritisk fra dokarkiv 😅 Skal fjerne 🧹